jdaviz/core/helpers.py

```diff
@@ -410,17 +410,23 @@ def show_in_new_tab(self, title=None):  # pragma: no cover
                       DeprecationWarning)
         return self.show(loc="sidecar:tab-after", title=title)
 
-    def _get_data(self, data_label=None, cls=None, subset_to_apply=None, function=None):
+    def _get_data(self, data_label=None, spatial_subset=None, spectral_subset=None,
+                  function=None, cls=None):
```

Comment:
@bmorris3 - how will this affect the downstream implementation of `get_data` in `lcviz`?

Reply:
We'll have to tweak LCviz a bit, and we may need a small tweak in this PR too. 

In https://github.com/spacetelescope/jdaviz/pull/2168, I added support for temporal subsets in `Application.get_data_from_viewer`. It looks like that may still work as long as this PR will return remaining subsets with `spatial_subset==False and spectral_subset==False` or `spatial_subset is None and spectral_subset is None`. Is that possible, @javerbukh? 

The LCviz [get_data "implementation"](https://github.com/spacetelescope/lcviz/blob/main/lcviz/helper.py#L108) (if you can call it that) is very straightforward to edit.

Reply:
`app.get_data_from_viewer` will be deprecated in the next two releases. I haven't gotten to that ticket yet (deprecating that and `get_subsets_from_viewer`) but I would not add new functionality there.

What do you mean by "return remaining subsets"?

Reply:
Since there are now also temporal subsets, in addition to spatial and spectral, we need to handle the case when `spatial_subset` and `spectral_subset` are False, but other `*al_subset`s are True. I'm thinking `temporal` here, but who knows if we'll have another case in the future. 

I was suggesting a "cheat" for supporting non-spatial/spectral subsets by allowing both kwargs to be set to False, and returning any non-spatial/spectral subsets. But maybe we should add `temporal_subset=None` to the call signature and handle it correctly.

Reply:
Going to resolve this conversation, if there are further thoughts please let me know!

Reply:
Is there a plan to go forward though?  We can definitely make updates to lcviz for this PR, I just want to make sure that this doesn't remove the flexibility that is needed.